coding_standard.md

```diff
@@ -0,0 +1,64 @@
+# Coding Standards
+
+## Formatting
+
+- Refer to the [clang-format configuration file](https://github.com/xEmbeddedTools/xmcu/blob/main/clang-format.txt) for the base standard.
+- Prioritize readability over adherence to the clang-format configuration when conflicts arise.
+
+## Names
+
+### Snake Case
+- Write all names in snake_case (lowercase letters with underscores separating words).
+
+### Argument Names
+- Start all argument names with `a_`.
+
+### Pointer Names
+- Start all pointer names with `p_`.
+
+### Global Variables/Constants
+- Start all global variables/constants with `g_`.
+
+### File-Scoped Variables/Constants
+- Start all variables/constants that are global within a file with `f_`.
+
+### Class/Struct/Union Names
+- Start all class, struct, or union names with a capital letter if there is a possibility to create an object of this type.
+- Start all other case names with a lowercase letter.
+
+## Class/Struct/Union
+
+### Access Specifiers
+- Use only one section per access specifier (`public`, `protected`, `private`).
+- Order access specifiers as follows: `public` first, `protected` second, `private` last.
+
+### Rule of 5 or Rule of 0
+- Follow the Rule of 5 or the Rule of 0 for special member functions.
+
+### Member Naming
+- Do not use `m_` prefix for members. Use `this->` to reference members.
+
+### Initialization List
+- Maintain the same order in the initialization list as in the declaration in the class.
+- Do not use members to create another in the initialization list.
+
+## Standard Library Usage
+
+### Use of Standard Libraries
+- Prefer standard libraries over custom implementations for common functionalities to ensure code reliability and maintainability.
+- Include necessary headers for the standard libraries you are using.
+- Avoid using deprecated standard library features.
+- Do not use `using namespace std;`.
+- Use `std` types for `cstdint` or `stdint.h`.
```

Comment:
Good point. The question is whether we should allow only `cstdint` and require all types to be used with the `std::` prefix, or allow only `stdint.h` so that these types can be used without the `std::` prefix, or both.

Personally, I prefer `int32_t` over `std::int32_t`, so I would choose to use only `stdint.h`. This approach does not violate the rule "Do not use using namespace std;."